packages/ui-kit/src/components/code-block.js

```diff
@@ -70,6 +70,42 @@ const TooltipBodyComponent = styled.div`
   font-size: ${typography.fontSizes.extraSmall};
   padding: ${dimensions.spacings.xs} ${dimensions.spacings.s};
 `;
+const languagesBackgroundImageUrl = color => {
+  return `url("data:image/svg+xml,%3C%3Fxml version='1.0' encoding='UTF-8'%3F%3E%3Csvg width='24px' height='24px' viewBox='0 0 24 24' version='1.1' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink'%3E%3C!-- Generator: sketchtool 45.2 (43514) - http://www.bohemiancoding.com/sketch --%3E%3Ctitle%3Ecaret-down%3C/title%3E%3Cdesc%3ECreated with sketchtool.%3C/desc%3E%3Cdefs%3E%3C/defs%3E%3Cg id='Icons' stroke='none' stroke-width='1' fill-rule='evenodd'%3E%3Cg id='MC-icon-set' transform='translate(-168.000000, -936.000000)' fill='%23${color}'%3E%3Cg id='Directions' transform='translate(24.000000, 888.000000)'%3E%3Cg id='Caret-Down' transform='translate(144.000000, 48.000000)'%3E%3Cpath d='M20.6658731,7.4053255 C20.4433682,7.16948908 20.1796129,7.05166867 19.8748538,7.05166867 L4.12508466,7.05166867 C3.82020235,7.05166867 3.55663185,7.16948908 3.33394217,7.4053255 C3.11125249,7.64142273 3,7.92055342 3,8.24323919 C3,8.56585976 3.11125249,8.84499045 3.33394217,9.08089208 L11.2088575,17.4207121 C11.4317935,17.6565485 11.695364,17.7746297 12,17.7746297 C12.304636,17.7746297 12.5684528,17.6565485 12.7909578,17.4207121 L20.6658731,9.08082687 C20.8883165,8.84499045 21,8.56585976 21,8.24317399 C21,7.92055342 20.8883165,7.64142273 20.6658731,7.4053255 L20.6658731,7.4053255 Z' id='shape'%3E%3C/path%3E%3C/g%3E%3C/g%3E%3C/g%3E%3C/g%3E%3C/svg%3E")`;
+};
+const LanguagesDropDown = styled.select`
```

Comment:
Could not find a good option.

Reply:
How is the design?

Reply:
https://commercetools.invisionapp.com/d/main?origin=v7#/console/17784193/388242717/preview?scrollOffset=1767

Reply:
Are we sure that those designs are still "valid/up-to-date"? Where is the behavior of the dropdown defined?

I don't understand why we need to create yet another component.

Reply:
You can find dropdown here:

https://commercetools.invisionapp.com/d/main?origin=v7#/console/17784193/388242718/preview?scrollOffset=1919

Reply:
@nkuehn , @Shecki , is there any change to the design of the dropdown in the CodeBlock header for multiple lamnguages?

Reply:
@emmenko  it was an explicit decision in the design process to go for the native dropdown and not the UI kit.  Let's not re-do the design process please.

Reply:
I agree to use the native dropdown, but the way it's styled now it doesn't feel/look like it.

![2020-03-18 08 53 08](https://user-images.githubusercontent.com/1110551/76937982-1dd58f00-68f6-11ea-98f9-24e626cfbc18.gif)

My main concern here is to avoid doing this sorts of implementation hacks.

I'm sure you all did your research but I don't feel comfortable with the way it is now.

---
For example, this is how Stripe uses the "native" HTML select:

![2020-03-18 08 57 12](https://user-images.githubusercontent.com/1110551/76938262-9e948b00-68f6-11ea-9996-40cd1881d7e6.gif)

It's still slightly customized but it feels more "native".

Another interesting option to consider is by having a "page switcher", like Prisma does

![2020-03-18 09 02 02](https://user-images.githubusercontent.com/1110551/76938656-4ca03500-68f7-11ea-97db-f241a131e69c.gif)

Or the more classic example with tabs. The advantage here is that you see all the possible languages, which might not be clear from the dropdown and people might not realise that there is one.

![2020-03-18 09 04 03](https://user-images.githubusercontent.com/1110551/76938866-ab65ae80-68f7-11ea-87e1-a85183fdb155.gif)

---

With that said, I apologize to bring this topic up again as you already did work on that, but I just don't feel comfortable with the way is currently designed/implemented.

Reply:
Hi @emmenko ,  I appreciate your efforts but can confirm it was explicitly decided in the design process to go more like the stripe example you posted . Aspects I remember:
 * should take little space in the header bar because there are lots of these and we intend to rather globally remember the state what language was last selected.  -> no tabs, rather use the space for a title with meaning
 * native select when in focus / opened is superior on mobile and "just works" for users without distracting
 * no explicit login, just remember the state in the browser. 

The fact that the dropdown is not native but weirdly styled in some browsers is a bug, I'll push the fix in a minute (background needs to be set in a wrapper and not the select itself - https://scottaohara.github.io/a11y_styled_form_controls/src/select/ ) 

I'm fine tuning the visibility of the little triangle to be visually stronger as part of post-launch optimization - it's in fact pretty easy to oversee.

Reply:
PS @davifantasia  you don't need to manually put moz / webkit prefixes when using emotion, it's doing that magic for us.

Reply:
P.P.S: To make it look the next bit better I just implemented a mapping to the most popular languages' actual names for display.   Will push in a minute.

Reply:
If it's supposed to "look native", why do we have the custom triangle?

Sorry to stress this topic but I want to avoid building something that is like a "workaround". If we're aiming for something like Stripe has, let's just do that. Keep it simple and "customize/hack" it too much.

Reply:
@emmenko  Stripe's design also has a fully custom styled non-expanded style and the native style when expanded.  That#'s what was designed and implemented here. 

I ask you all to stop this discussion here and make it part of a design review after launch, including feedback from actual users.  We're just throwing our internal  opinions in circles here.

Reply:
Maybe you misunderstood my argument. My main concern here is about keeping the implementation maintainable by following our design system rules.
It's my job as a developer to ensure that and to point that out. From my experience design decisions are sometimes inconsistent (not intentionally) and we need to make sure that we follow our own defined rules. Which means for example to point these things out to the designers or whoever is involved. We shouldn't take everything for granted.

I understand that you both are not used to this but this is important as we are/want to build quality software (especially when it's open sourced).

I'm willing to stop this conversation here, as this is not a super important thing, but I ask you that we follow this up.

Thanks for trying to understand my point of view.